src/sentry/eventstore/snuba/backend.py

```diff
@@ -92,26 +94,34 @@ def __get_events(
         offset=DEFAULT_OFFSET,
         referrer=None,
         should_bind_nodes=False,
+        event_list=None,
     ):
         assert filter, "You must provide a filter"  # NOQA
         cols = self.__get_columns()
         orderby = orderby or DESC_ORDERING
+        if event_list is not None and not filter.event_ids:
+            filter.event_ids = [event.event_id for event in event_list]
 
         # This is an optimization for the Group.filter_by_event_id query where we
         # have a single event ID and want to check all accessible projects for a
         # direct hit. In this case it's usually faster to go to nodestore first.
         if (
             filter.event_ids
             and filter.project_ids
-            and len(filter.event_ids) * len(filter.project_ids) < min(limit, NODESTORE_LIMIT)
+            and (
+                len(filter.event_ids) * len(filter.project_ids) < min(limit, NODESTORE_LIMIT)
```

Comment:
so with trace-view, we can get up to 100 event_ids, and a large number of project_ids, but we'll ever only want to get up to 100 events, staying below this limit.
But since this is creating a list of events for every project + event we'll usually go over this limit and miss this optimization.